### PR TITLE
docs: how-to for IDE setup [DET-2755]

### PR DESCRIPTION
**Pull request by davetroiano**
## Description

docs: how-to for IDE setup

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

### CODE REVIEW

File: docs/how-to/ide-setup.txt [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+the TensorFlow or PyTorch library versions that your code will run
+against when submitting to the cluster. For example, if you are using
+Determined's `default task container image
+<https://github.com/determined-ai/environments/blob/master/Makefile>`__
+for a PyTorch model, you would install the following dependencies in
+your virtualenv:
+
+.. code::
+
+   pip install 'determined[tensorflow]'==<Determined version>
```
Comment (neilconway) [inline docs/how-to/ide-setup.txt:36]: If we want to use PyTorch, why `determined[tensorflow]`?

File: docs/how-to/ide-setup.txt [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+   pip install torch==<Determined's default image PyTorch version, e.g., 1.4.0>
+   pip install torchvision==<Determined's default image torchvision version, e.g., 0.5.0>
+
+If you are using a :ref:`customer Docker image <custom-docker-images>`
```
Comment (neilconway) [inline docs/how-to/ide-setup.txt:40]: "customer" or "custom"?

File: docs/how-to/ide-setup.txt [hunk 3]
```diff
@@ -0,0 +1,112 @@
+.. _ide-setup:
+
+#####################
+ IDE Setup (PyCharm)
+#####################
+
+Model developers may develop and debug models with their preferred IDE.
+In this guide, we show how to configure PyCharm to develop and debug
+against Determined's :ref:`Native API <tutorials_native-api>`.
```
Comment (neilconway) [inline docs/how-to/ide-setup.txt:9]: Can we add a bit of clarifying language about why the native API is different than the trial API in this respect / what you need to do (nothing?) if using the trial API.

Comment (davetroiano) [reply docs/how-to/ide-setup.txt]: yes I took a pass at clarifying this

File: docs/how-to/ide-setup.txt (hunk 2 above)
Comment (davetroiano) [reply docs/how-to/ide-setup.txt]: fixed - "custom"

File: docs/how-to/ide-setup.txt (hunk 1 above)
Comment (davetroiano) [reply docs/how-to/ide-setup.txt]: removed TF as it's not needed here

**Review by neilconway — APPROVED** [review]
